server/app/Models/Doc.php

```diff
@@ -230,67 +220,45 @@ public function getPagesAttribute()
         return $this->getPages();
     }
 
-    public function getCommentCount()
-    {
-        return $this->comments()->count();
-    }
-
     public function getCommentCountAttribute()
     {
-        return $this->getCommentCount();
+        // note that this is only the top level comment count
+        return $this->comments
+            ->filter(function ($item) {
+                return !$item->isNote();
+            })
+            ->count()
+            ;
```

Comment:
`filter` is a really inefficient to achieve these ends - we're getting everything from the database and then filtering in memory in PHP.  Ideally we'd do this via the database query.